tensorflow_addons/seq2seq/tests/basic_decoder_test.py

```diff
@@ -319,274 +319,243 @@ def test_step_with_sample_embedding_helper():
-                    "step_next_inputs": step_next_inputs,
-                    "step_finished": step_finished,
-                }
-            )
+    eval_result = {
+        "batch_size": batch_size_t,
+        "first_finished": first_finished,
+        "first_inputs": first_inputs,
+        "first_state": first_state,
+        "step_outputs": step_outputs,
+        "step_state": step_state,
+        "step_next_inputs": step_next_inputs,
+        "step_finished": step_finished,
+    }
 
-            self.assertAllEqual(
-                [False, False, False, False, True], eval_result["first_finished"]
-            )
-            self.assertAllEqual(
-                [False, False, False, True, True], eval_result["step_finished"]
-            )
-            sample_ids = eval_result["step_outputs"].sample_id
-            self.assertEqual(output_dtype.sample_id, sample_ids.dtype)
-            batch_where_not_sampling = np.where(sample_ids == -1)
-            batch_where_sampling = np.where(sample_ids > -1)
-            self.assertAllClose(
-                eval_result["step_next_inputs"][batch_where_sampling],
-                embeddings[sample_ids[batch_where_sampling]],
-            )
-            self.assertAllClose(
-                eval_result["step_next_inputs"][batch_where_not_sampling],
-                np.squeeze(inputs[batch_where_not_sampling, 1], axis=0),
-            )
+    np.testing.assert_equal(
+        np.asanyarray([False, False, False, False, True]),
+        eval_result["first_finished"].numpy(),
+    )
+    np.testing.assert_equal(
+        np.asanyarray([False, False, False, True, True]),
+        eval_result["step_finished"].numpy(),
+    )
+    sample_ids = eval_result["step_outputs"].sample_id
+    assert output_dtype.sample_id == sample_ids.dtype
+    batch_where_not_sampling = tf.cast(np.where(sample_ids == -1), dtype=tf.int32)
+    batch_where_sampling = tf.cast(np.where(sample_ids > -1), dtype=tf.int32)
 
-    def _testStepWithScheduledOutputTrainingHelper(
-        self, sampling_probability, use_next_inputs_fn, use_auxiliary_inputs
-    ):
-        sequence_length = [3, 4, 3, 1, 0]
-        batch_size = 5
-        max_time = 8
-        input_depth = 7
-        cell_depth = input_depth
-        if use_auxiliary_inputs:
-            auxiliary_input_depth = 4
-            auxiliary_inputs = np.random.randn(
-                batch_size, max_time, auxiliary_input_depth
-            ).astype(np.float32)
-        else:
-            auxiliary_inputs = None
+    np.testing.assert_equal(
+        tf.squeeze(
+            tf.gather(eval_result["step_next_inputs"], batch_where_sampling)
+        ).numpy(),
+        tf.squeeze(tf.gather(embeddings[sample_ids, :], batch_where_sampling)).numpy(),
+    )
+    np.testing.assert_equal(
+        tf.squeeze(
+            tf.gather(eval_result["step_next_inputs"], batch_where_not_sampling)
+        ),
+        np.squeeze(inputs[batch_where_not_sampling, 1], axis=0),
+    )
```

Comment:
Here could you detail a bit more why you did the changes? It's not clear to me.

Reply:
The original slice method can't pass the testing until the changes, i.e., dimension inconsistent.

Reply:
Why did the tests passed before then? Were the test functions not run? Or is it because it's in eager mode now?